prov/gni/src/gnix_msg.c

```diff
@@ -76,13 +86,7 @@ static int __gnix_recv_completion(struct gnix_fid_ep *ep,
 	cq = ep->recv_cq;
 	assert(cq != NULL);
 
-	rc = _gnix_cq_add_event(cq,
-				req->user_context,
-				FI_RECV | FI_MSG,
-				req->msg.recv_len,
-				(void *)req->msg.recv_addr,
-				req->msg.imm,
-				0);
+	rc = _gnix_cq_add_event(cq, context, flags, len, addr, data, tag);
```

Comment:
The request was modified before providing the information. In the case of a peek, you don't want to modify the req, so a wrapper function from which the values could be passed in directly was made.